### PR TITLE
Split texture upload performance test into own `TestScene`

### PR DESCRIPTION
**Pull request by peppy**
This also makes the uploads happen every frame, as this is more useful for diagnosing contention edge cases.

Notes from testing:

- Metal can show signs of corruption with higher texture upload counts.
- D3D has very low performance compared to metal with similar texture upload counts.

### CODE REVIEW

File: osu.Framework.Tests/Visual/Performance/TestSceneTextureUploadPerformance.cs [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+
+        protected override void LoadComplete()
+        {
+            base.LoadComplete();
+
+            AddSliderStep("count", 1, 100, 10, v =>
+            {
+                count = v;
+                recreate();
+            });
```
Comment (frenzibyte) [inline osu.Framework.Tests/Visual/Performance/TestSceneTextureUploadPerformance.cs:47]: Right now changing the value of the slider recreates the fill flow and all sprites, causing a flicker on every instant value when "sliding" the slider.

Creating the flow once and adding/removing sprites based on the count number (similar to `TestSceneBoxPerformance`) would feel better:
```diff
diff --git a/osu.Framework.Tests/Visual/Performance/TestSceneTextureUploadPerformance.cs b/osu.Framework.Tests/Visual/Performance/TestSceneTextureUploadPerformance.cs
index 6c077fc93..a0933b408 100644
--- a/osu.Framework.Tests/Visual/Performance/TestSceneTextureUploadPerformance.cs
+++ b/osu.Framework.Tests/Visual/Performance/TestSceneTextureUploadPerformance.cs
@@ -19,8 +19,6 @@ namespace osu.Framework.Tests.Visual.Performance
 {
     public partial class TestSceneTextureUploadPerformance : PerformanceTestScene
     {
-        private int count;
-
         private FillFlowContainer<Sprite>? fill;
 
         [Resolved]
@@ -40,15 +38,6 @@ protected override void LoadComplete()
         {
             base.LoadComplete();
 
-            AddSliderStep("count", 1, 100, 10, v =>
-            {
-                count = v;
-                recreate();
-            });
-        }
-
-        private void recreate()
-        {
             Child = fill = new FillFlowContainer<Sprite>
             {
                 RelativeSizeAxes = Axes.X,
@@ -58,16 +47,22 @@ private void recreate()
                 Origin = Anchor.Centre,
             };
 
-            for (int i = 0; i < count; i++)
+            AddSliderStep("count", 1, 100, 10, newCount =>
             {
-                fill.Add(new Sprite
-                {
-                    Size = new Vector2(128),
-                    Texture = renderer.CreateTexture(512, 512)
-                });
-            }
+                for (int i = fill.Count - 1; i >= newCount; i--)
+                    fill.Remove(fill.Children[i], true);
+
+                for (int i = fill.Count; i < newCount; i++)
+                    fill.Add(createSprite());
+            });
         }
 
+        private Sprite createSprite() => new Sprite
+        {
+            Size = new Vector2(64),
+            Texture = renderer.CreateTexture(512, 512)
+        };
+
         private ulong lastUploadedFrame;
 
         protected override void Update()

```

Comment (peppy) [reply osu.Framework.Tests/Visual/Performance/TestSceneTextureUploadPerformance.cs:47]: This is the same as every other performance test scene though, right?

The plan is to go through and fix all of these in one go.

Comment (peppy) [reply osu.Framework.Tests/Visual/Performance/TestSceneTextureUploadPerformance.cs:47]: Feel free to commit the change, just note that it needs to be fixed everywhere else.

**Review by frenzibyte — APPROVED** [review]
